packages/stream/solana/distributor/client.ts

```diff
@@ -0,0 +1,268 @@
+import BN from "bn.js";
+import {
+  ASSOCIATED_TOKEN_PROGRAM_ID,
+  NATIVE_MINT,
+  TOKEN_PROGRAM_ID,
+  createTransferCheckedInstruction,
+  getMint,
+} from "@solana/spl-token";
+import {
+  Connection,
+  PublicKey,
+  SystemProgram,
+  TransactionInstruction,
+  Transaction,
+  Commitment,
+  ConnectionConfig,
+} from "@solana/web3.js";
+
+import { DISTRIBUTOR_PROGRAM_ID } from "./constants";
+import { ICluster, ITransactionResult } from "../../common/types";
+import {
+  IClaimData,
+  IClawbackData,
+  ICreateDistributorData,
+  ICreateDistributorResult,
+  IGetClaimsData,
+  IGetDistributors,
+} from "./types";
+import { ICreateSolanaExt, IInteractStreamSolanaExt } from "../types";
+import { ata, checkOrCreateAtaBatch } from "../utils";
+import {
+  ClaimLockedAccounts,
+  ClawbackAccounts,
+  NewClaimAccounts,
+  NewClaimArgs,
+  NewDistributorAccounts,
+  NewDistributorArgs,
+  claimLocked,
+  clawback,
+  newClaim,
+  newDistributor,
+} from "./generated/instructions";
+import { prepareWrappedAccount } from "../instructions";
+import { ClaimStatus, MerkleDistributor } from "./generated/accounts";
+import { getClaimantStatusPda, getDistributorPda, wrappedSignAndExecuteTransaction } from "./utils";
+
+interface IInitOptions {
+  clusterUrl: string;
+  cluster?: ICluster;
+  commitment?: Commitment | ConnectionConfig;
+  programId?: string;
+}
+
+export default class SolanaDistributorClient {
+  private connection: Connection;
+
+  private programId: PublicKey;
+
+  private commitment: Commitment | ConnectionConfig;
+
+  /**
+   * Create Stream instance
+   */
+  constructor({
+    clusterUrl,
+    cluster = ICluster.Mainnet,
+    commitment = "confirmed",
+    programId = "",
+  }: IInitOptions) {
+    this.commitment = commitment;
+    this.connection = new Connection(clusterUrl, this.commitment);
+    this.programId =
+      programId !== "" ? new PublicKey(programId) : new PublicKey(DISTRIBUTOR_PROGRAM_ID[cluster]);
+  }
+
+  public async create(
+    data: ICreateDistributorData,
+    { sender, isNative = false }: ICreateSolanaExt
+  ): Promise<ICreateDistributorResult> {
+    if (!sender.publicKey) {
+      throw new Error("Sender's PublicKey is not available, check passed wallet adapter!");
+    }
+
+    const ixs: TransactionInstruction[] = [];
+    const mint = isNative ? NATIVE_MINT : new PublicKey(data.mint);
+    const mintAccount = await getMint(this.connection, mint);
+    const distributorPublicKey = getDistributorPda(this.programId, mint, data.version);
+    const tokenVault = await ata(mint, distributorPublicKey);
+    const senderTokens = await ata(mint, sender.publicKey);
```

Comment:
can we somehow make these requests in parallel? 
All this RPC calls can be slow as we see always 😄